tests/ci/run_m1_ec2_instance.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+set -exo pipefail
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# Please run from project root folder!
+# You'll want to set the codebuild env variables set if running locally
+
+# cleanup code
+cleanup() {
+  set +e
+  # delete the various documents that we created
+  for name in ${ssm_document_names};do
+    aws ssm delete-document --name "${name}"
+  done
+}
+
+# we wanna run the cleanup code on exit
+trap cleanup EXIT
+
+# print some information for reference
+echo GitHub PR Number: "${CODEBUILD_WEBHOOK_TRIGGER}"
+echo GitHub Branch Name: "${CODEBUILD_WEBHOOK_HEAD_REF}"
+AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)
+echo AWS Account ID: "${AWS_ACCOUNT_ID}"
+echo GitHub Repo Link: "${CODEBUILD_SOURCE_REPO_URL}"
+export CLOUDWATCH_LOG_GROUP="aws-lc-ci-macos-arm-cw-logs"
+
+# get information for ec2 instances
+vpc_id="$(aws ec2 describe-vpcs --filter Name=tag:Name,Values=aws-lc-ci-bm-framework/aws-lc-ci-bm-framework-ec2-vpc --query Vpcs[*].VpcId --output text)"
+sg_id="$(aws ec2 describe-security-groups --filter Name=vpc-id,Values="${vpc_id}" --filter Name=group-name,Values=bm_framework_ec2_sg --query SecurityGroups[*].GroupId --output text)"
+subnet_id="$(aws ec2 describe-subnets --filter Name=vpc-id,Values="${vpc_id}" --filter Name=state,Values=available --filter Name=tag:Name,Values=aws-lc-ci-bm-framework/aws-lc-ci-bm-framework-ec2-vpc/PrivateSubnet1 --query Subnets[*].SubnetId --output text)"
+ec2_instance="$(aws ec2 describe-instances --filter Name=tag:Name,Values=aws-lc-ci-macos-arm-ec2-instance --query Reservations[*].Instances[*].InstanceId --output text)"
+
+generate_ssm_document_file() {
```

Comment:
Maybe not in the scope of this PR, but I recognize a lot of these helper functions (e.g. `generate_ssm_document_file`, `create_ec2_instance`, etc.) from the bash script for the benchmarking framework. Would you say it's worth finding a way to refactor the scripts so that commonly used helper aren't duplicated?